web/modules/custom/bnf/src/Services/BnfImporter.php

```diff
@@ -37,26 +46,26 @@ public function __construct(
   ) {}
 
   /**
-   * Get node title from BNF.
+   * Get node meta data from BNF.
    */
-  public function getNodeTitle(string $uuid, string $endpointUrl): string {
+  public function getNodeMetaData(string $uuid, string $endpointUrl): NodeArticleMetaData|NodePageMetaData|NodeGoArticleMetaData|NodeGoCategoryMetaData|NodeGoPageMetaData {
     $endpointConfig = new SailorEndpointConfig(MangleUrl::server($endpointUrl));
-    Configuration::setEndpointFor(GetNodeTitle::class, $endpointConfig);
-    $response = GetNodeTitle::execute($uuid);
+    Configuration::setEndpointFor(GetNodeMetaData::class, $endpointConfig);
+    $response = GetNodeMetaData::execute($uuid);
 
     $nodeData = $response->data?->node;
 
     if (!$nodeData) {
       throw new \RuntimeException('Could not fetch content.');
     }
 
-    return $nodeData->title;
+    return $nodeData;
   }
 
   /**
    * Importing a node from a GraphQL source endpoint.
    */
-  public function importNode(string $uuid, string $endpointUrl, string $nodeType = 'article'): NodeInterface {
+  public function importNode(string $uuid, string $nodeType, string $endpointUrl): NodeInterface {
     if (!in_array($nodeType, self::ALLOWED_CONTENT_TYPES)) {
```

Comment:
You've extendend an API, intrduced five classes and spent effort carrying the type through forms to importNode, only for this check, other than that, it's just used for logging that could live without.

But we can't import a node without a mapper, and the mapper manager will throw an exception if it can't find a mapper for the GraphQL object fetched. And the GraphQL class is implicitly bound to the node type. So it'll only map content types we "allow" by actually implementing them.

So this check, and the code supporting it, is not needed.

Reply:
jesus christ, you're right.. i think it's because the importer *used* to need the content type, before you did the big overhaul, and i never realized until now

nghh.. i'll fix it.

Reply:
Fixed here: https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/2267/commits/be9d7ef535ea9d03aaa34bb56d8dc36b285ac47a